pymc/distributions/mixture.py

```diff
@@ -27,28 +27,31 @@
 from pymc.distributions.continuous import Normal, get_tau_sigma
 from pymc.distributions.dist_math import check_parameters
 from pymc.distributions.distribution import (
-    Discrete,
-    Distribution,
     SymbolicDistribution,
     _get_moment,
     get_moment,
 )
 from pymc.distributions.logprob import logcdf, logp
 from pymc.distributions.shape_utils import to_tuple
 from pymc.util import check_dist_not_registered
-from pymc.vartypes import discrete_types
+from pymc.vartypes import continuous_types, discrete_types
 
 __all__ = ["Mixture", "NormalMixture"]
 
 
-def all_discrete(comp_dists):
+def all_same_type(comp_dists):
     """
-    Determine if all distributions in comp_dists are discrete
+    Determine if the distributions in comp_dists are all discrete or continuous
     """
-    if isinstance(comp_dists, Distribution):
-        return isinstance(comp_dists, Discrete)
+
+    if len(comp_dists) == 1:
+        # should return True for either check_discrete=True or False
+        return comp_dists[0] in continuous_types | discrete_types
     else:
-        return all(isinstance(comp_dist, Discrete) for comp_dist in comp_dists)
+        all_continuous = all([comp_dist in continuous_types for comp_dist in comp_dists])
```

Comment:
shouldn't it be based on the comp_dist.dtype?

Reply:
...yes, late night brain farts 😅